pkg/controller/privatelink/actuator/gcpactuator/gcplinkactuator.go

```diff
@@ -299,17 +321,42 @@ func (a *GCPLinkActuator) ensureServiceAttachmentSubnet(cd *hivev1.ClusterDeploy
 }
 
 func getServiceAttachmentSubnetCIDR(cd *hivev1.ClusterDeployment) string {
-	cidr := "192.168.0.0/24"
 	if cd != nil &&
 		cd.Spec.Platform.GCP != nil &&
 		cd.Spec.Platform.GCP.PrivateServiceConnect != nil &&
 		cd.Spec.Platform.GCP.PrivateServiceConnect.ServiceAttachment != nil &&
 		cd.Spec.Platform.GCP.PrivateServiceConnect.ServiceAttachment.Subnet != nil &&
 		cd.Spec.Platform.GCP.PrivateServiceConnect.ServiceAttachment.Subnet.Cidr != "" {
 
-		cidr = cd.Spec.Platform.GCP.PrivateServiceConnect.ServiceAttachment.Subnet.Cidr
+		return cd.Spec.Platform.GCP.PrivateServiceConnect.ServiceAttachment.Subnet.Cidr
+	}
+	return defaultServiceAttachmentSubnetCidr
+}
+
+func getServiceAttachmentSubnetExistingName(cd *hivev1.ClusterDeployment) string {
+	if cd != nil &&
+		cd.Spec.Platform.GCP != nil &&
+		cd.Spec.Platform.GCP.PrivateServiceConnect != nil &&
+		cd.Spec.Platform.GCP.PrivateServiceConnect.ServiceAttachment != nil &&
+		cd.Spec.Platform.GCP.PrivateServiceConnect.ServiceAttachment.Subnet != nil &&
+		cd.Spec.Platform.GCP.PrivateServiceConnect.ServiceAttachment.Subnet.Existing != nil {
+
+		return cd.Spec.Platform.GCP.PrivateServiceConnect.ServiceAttachment.Subnet.Existing.Name
+	}
+	return ""
+}
+
+func getServiceAttachmentSubnetExistingProject(cd *hivev1.ClusterDeployment) string {
+	if cd != nil &&
+		cd.Spec.Platform.GCP != nil &&
+		cd.Spec.Platform.GCP.PrivateServiceConnect != nil &&
+		cd.Spec.Platform.GCP.PrivateServiceConnect.ServiceAttachment != nil &&
+		cd.Spec.Platform.GCP.PrivateServiceConnect.ServiceAttachment.Subnet != nil &&
+		cd.Spec.Platform.GCP.PrivateServiceConnect.ServiceAttachment.Subnet.Existing != nil {
+
+		return cd.Spec.Platform.GCP.PrivateServiceConnect.ServiceAttachment.Subnet.Existing.Project
 	}
-	return cidr
+	return ""
```

Comment:
It's a shame that such a hack is required to write concise code

Reply:
Noted. I wasn't aware of this hack. I can do this if there arises a need to make another change (and therefore do another 3hr manual testing cycle on the PR).

Reply:
> I wasn't aware of this trick.

Heh, me neither, wasn't even sure it would work. I had thought of it in passing every time I saw this pattern, but finally decided to try it today. (Those playground links are mine.)

> It's a shame that such a hack is required to write concise code

It's not super different from try/except (or try/catch, or however it's spelled in your preferred language). Though definitely has a different feel, which takes some getting used to. Like channels and goroutines.

I'm curious how you would do it in python or perl or java or whatever?

Reply:
In my experience, most languages where null pointer exceptions are possible would require what I did here by walking the chain.

Reply:
Or for example

```python
try:
  return thing.field.otherfield.whatever
except NullPointerException:
  return default
```

The only improvement here is being specific about what error we expect, allowing any other error to continue to panic. It may be possible for golang's `recover()` to be more specific similarly -- haven't played with it.